pkg/koordlet/resmanager/cpu_suppress.go

```diff
@@ -182,29 +176,37 @@ func calculateBESuppressCPUSetPolicy(cpusetQuantity *resource.Quantity, oldCPUSe
 	// NOTE: here we prioritize the processors by the affinity of numa node/socket and hyper-threading so that the
 	//       cpuset cpus can be picked in order, rather than scoring all the possible policies.
 	// TBD: may just use a lookup table for a steady state
-
-	// initially sorted by processor id
-	prioritizedCPUs := nodeCPUInfo.ProcessorInfos
+	prioritizedCPUs := processorInfos
 	sort.Slice(prioritizedCPUs, func(i, j int) bool {
 		return prioritizedCPUs[i].CPUID < prioritizedCPUs[j].CPUID
 	})
 
 	// getNodeIndex is a function to calculate an index for every numa node or socket
 	getNodeIndex := func(info util.ProcessorInfo) int32 {
+		// (nodeId, socketId) => nodeIndex
 		return (info.NodeID + numProcessors) * (info.SocketID + 1)
 	}
 
 	// get the relative indexes of the processor on its node
-	indexOfNode := map[int32]int32{}
-	numProcessorsOfNode := map[int32]int32{}
+	coreIndexOfNode := map[int32]int32{}
+	coreIdSetofNode := map[int32]map[int32]int32{}
 	for _, processor := range prioritizedCPUs {
-		numProcessorsOfNode[getNodeIndex(processor)]++
-		if indexOfNode[processor.CoreID] <= 0 {
-			indexOfNode[processor.CoreID] = numProcessorsOfNode[getNodeIndex(processor)]
+		nodeIndex := getNodeIndex(processor)
+		if _, ok := coreIdSetofNode[nodeIndex]; !ok {
```

Comment:
`coreIdSetofNode` may be incorrect when two CPUs are on different Sockets but have the same Core ID.